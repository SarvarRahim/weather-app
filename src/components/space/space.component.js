import React from 'react';
import { hoc } from '@utils/hoc';
import { useSpaceProps } from './space.props';
import { SpaceContainer } from './space.style';

/**
 * <Space />
 */
const Space = hoc(useSpaceProps, (props) => (
  <SpaceContainer {...props} />
));

export { Space };
