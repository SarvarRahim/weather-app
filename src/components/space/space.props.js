
/**
 * <Space /> props
 */
export const useSpaceProps = (_) => {};

