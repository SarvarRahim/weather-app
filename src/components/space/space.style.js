import styled from 'styled-components'

export const SpaceContainer = styled.div`
  padding-top: ${props => props.space || 0};
`