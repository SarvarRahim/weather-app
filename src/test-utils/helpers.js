import { render } from '@testing-library/react';
import React, { useRef } from 'react';
import { BrowserRouter, Router } from 'react-router-dom';
import { Provider as ReduxProvider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import { theme } from '@styles/Theme';
import { store } from '@store';
import { Formik } from 'formik';
import { VideoContext } from '@courses/routes/courses-single/course.context';
import { createMemoryHistory } from 'history';

const AllTheProviders = ({ children, formikValues = {} }) => {
  const videoRef = useRef();
  return (
    <BrowserRouter>
      <ReduxProvider store={store}>
        <ThemeProvider theme={theme}>
          <Formik initialValues={formikValues}>
            <VideoContext.Provider value={videoRef}>
              {children}
            </VideoContext.Provider>
          </Formik>
        </ThemeProvider>
      </ReduxProvider>
    </BrowserRouter>
  );
};

export const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export const renderWithRouter = (
  ui,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...render(<Router history={history}>{ui}</Router>),
    history
  };
};
