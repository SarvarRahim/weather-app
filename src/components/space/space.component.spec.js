import { customRender } from '@test-utils/helpers';
import { Space } from './space.component';

test('Should render Space component', () => {
  const { asFragment } = customRender(<Space space='15px' />);

  expect(asFragment).toMatchSnapshot();
});
