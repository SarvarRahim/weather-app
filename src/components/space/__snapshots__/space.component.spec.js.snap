// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Space component 1`] = `[Function]`;
